.github/actions/preparation/action.yml

```
@@ -0,0 +1,25 @@
+name: "Preparation"
+description: "Preparation steps to run tests"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout source
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+    - name: Install poetry
+      uses: abatilo/actions-poetry@v2
+    - name: Setup a local virtual environment (if no poetry.toml file)
```

```suggestion
    - name: Setup a local virtual environment (if no poetry.lock file)
```

Here I'm sceptical. There is always a poetry.lock file present. Is this step necessary? What does it do?

---

I don't know the set-up well enough to judge this.

---

I copy-pasted this from the action's docs, seems you could provide a `poetry.toml` file to configure these things (which we don't do) that would override this. But this specifies that the virtual environment is created whichin the repo, so that it is accessible to the caching action.

Again, I don't really understand too much of this, I just copied it from the action's docs. But to be able to test it, we need to merge it

---

If you want we can make YOLO and just merge it, and then see if there are problems.

But what is the purpose of this PR? I suppose you want to factor out these steps to economize time. But I argue that these steps are very fast anyways, because GitHub CI provides us with pre-existing ubuntu and python. So we could also try to repeat these steps every time.

---

It is more a code-organisational question that I want to factor this out: If I make everything a separate job, I would have to repeat this too many times for my taste. So I thought I'd follow the pattern we have in the API repo instead.

But I will remove the premade poetry action and go back to what there is currently... that would be less YOLO 😉